intake/source/csv.py

```diff
@@ -54,12 +54,19 @@ def __init__(self, urlpath, csv_kwargs=None, metadata=None,
     def _set_pattern_columns(self, path_column):
         """Get a column of values for each field in pattern
         """
+        try:
+            from pandas.api.types import CategoricalDtype
+            _HAS_CDT = True
+        except ImportError:
+            _HAS_CDT = False
+
         col = self._dataframe[path_column]
         paths = col.cat.categories
 
         column_by_field = {field:
-            col.cat.codes.map(dict(enumerate(values))).astype("category") for
-            field, values in reverse_formats(self.pattern, paths).items()
+            col.cat.codes.map(dict(enumerate(values))).astype(
+                "category" if not _HAS_CDT else CategoricalDtype(set(values))
+            ) for field, values in reverse_formats(self.pattern, paths).items()
```

Comment:
So the original type loaded by Dask is still string, but we convert to get categoricals at the next stage?

Reply:
no sure I understand the distinction.

Reply:
I think I was misunderstanding what happens here. Is the following right:
upon load, dask applies the filename pieces to each chunk, to create categorical columns which always have just a  single label within each partition; then in this step we change the labels to real values based on the parsing. Is that right? In that case, everything is fine.